### PR TITLE
[14.0][IMP] account_invoice_inter_company: added multi company tab 

### PR DESCRIPTION
**Pull request by ilyasProgrammer**
Added multi company tab in the invoice to see related counterpart invoice.

### CODE REVIEW

**Review by francesco-ooops — APPROVED** [review]
Functional review ok!

in this PR we introduce an "intercompany info" tab in invoice to provide a direct link between customer invoice and related vendor bill for other company.

We opted for a tab instead of a smart button as it can contain further information about the related document, in this case: Source invoice delivery address (useful for searching or group by)

**Review by elvise — APPROVED** [review]
LGTM!

**Comment by francesco-ooops** [comment]
@OCA/intercompany-maintainers could you review? thanks!

**Comment by elvise** [comment]
@pedrobaeza what do you think is it ready for merge ?

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
The sale module mustn't be introduced as dependency

**Comment by elvise** [comment]
@ilyasProgrammer Done ?

**Comment by ilyasProgrammer** [comment]
> @ilyasProgrammer Done ?

yes

**Comment by ilyasProgrammer** [comment]
@pedrobaeza "sale" required to use partner_shipping_id field. Should we create separate module in this case ?

**Comment by pedrobaeza** [comment]
Yes if you want to use that field. The are s lot of systems with account, but no sale module installed, and you can't force them to install it just for that field.

**Comment by ilyasProgrammer** [comment]
@pedrobaeza can we merge this? I will create new PR with shipping address field.

File: account_invoice_inter_company/readme/CONTRIBUTORS.rst [hunk 1]
```diff
@@ -11,4 +11,8 @@
   * Jairo Llopis
   * David Vidal
   * Pedro M. Baeza
+
```
Comment (pedrobaeza) [inline account_invoice_inter_company/readme/CONTRIBUTORS.rst]: This empty line is not needed.

File: account_invoice_inter_company/models/account_move.py [hunk 2]
```diff
@@ -23,6 +23,12 @@ class AccountMove(models.Model):
         copy=False,
         prefetch=False,
     )
+    related_bill_id = fields.Many2one(
```
Comment (pedrobaeza) [inline account_invoice_inter_company/models/account_move.py:26]: This one is not filled for the past ones. Why not using the inverse link?

Comment (ilyasProgrammer) [reply account_invoice_inter_company/models/account_move.py:26]: @pedrobaeza I dont know how to use inverse here. related_bill_id  can be computed using search by auto_invoice_id. Is that what you mean ?

Comment (pedrobaeza) [reply account_invoice_inter_company/models/account_move.py:26]: Yeah, a computed or one2many.

Comment (elvise) [reply account_invoice_inter_company/models/account_move.py:26]: @ilyasProgrammer Done ?

**Review by francesco-ooops — APPROVED** [review]
functional ok

@pedrobaeza good for merge?

File: account_invoice_inter_company/models/account_move.py [hunk 3]
```diff
@@ -23,6 +23,22 @@ class AccountMove(models.Model):
...
+        string="Related Bill",
+        readonly=True,
+        copy=False,
+    )
+
+    def _compute_related_bill_id(self):
+        for rec in self:
+            rec.related_bill_id = (
+                rec.related_bill_ids[0].id if rec.related_bill_ids else False
+            )
```
Comment (pedrobaeza) [inline account_invoice_inter_company/models/account_move.py]:
```suggestion
            rec.related_bill_id = rec.related_bill_ids[:1].id
```

File: account_invoice_inter_company/readme/CONTRIBUTORS.rst [hunk 4]
```diff
@@ -12,3 +12,6 @@
   * David Vidal
   * Pedro M. Baeza
 * Isaac Gallart <igallart@puntsistemes.es>
+
+* `Ooops404 <https://www.ooops404.com>`__:
+  * Ilyas <irazor147@gmail.com>
```
Comment (pedrobaeza) [inline account_invoice_inter_company/readme/CONTRIBUTORS.rst]:
```suggestion
* `Ooops404 <https://www.ooops404.com>`__:

    * Ilyas <irazor147@gmail.com>
```

File: account_invoice_inter_company/views/account_move_views.xml [hunk 5]
```diff
@@ -13,4 +13,30 @@
...
+                            string="Invoice Info"
+                            attrs="{'invisible': [('auto_invoice_id', '=', False)]}"
+                        >
+                            <field name="auto_invoice_id" />
+                        </group>
+                        <group
+                            string="Bill Info"
+                            attrs="{'invisible': [('related_bill_id', '=', False)]}"
+                        >
+                            <field name="related_bill_id" />
```
Comment (pedrobaeza) [inline account_invoice_inter_company/views/account_move_views.xml]: Is this invoice of other company?

Comment (ilyasProgrammer) [reply account_invoice_inter_company/views/account_move_views.xml]: yes

Comment (pedrobaeza) [reply account_invoice_inter_company/views/account_move_views.xml]: Then you can't add it, or you will have access errors if you are not logged in both companies.

Comment (ilyasProgrammer) [reply account_invoice_inter_company/views/account_move_views.xml]: No. You just wont see it.

Comment (pedrobaeza) [reply account_invoice_inter_company/views/account_move_views.xml]: That's not how this works I'm afraid. Have you tried it?

Comment (ilyasProgrammer) [reply account_invoice_inter_company/views/account_move_views.xml]: https://recordit.co/Na7NKn04Ns

Comment (pedrobaeza) [reply account_invoice_inter_company/views/account_move_views.xml]: Uhm, I didn't know about that curious behavior, but then there's a misuse of the feature: if you are not logged in both companies, you may think that the invoices isn't is an inter-company one. I think you have to put 2 computed fields with `compute_sudo=True`:

- Related invoice number as text.
- Related company.

And you can replace the text by the many2one if the company is in allowed ones.

Comment (ilyasProgrammer) [reply account_invoice_inter_company/views/account_move_views.xml]: how about this: https://recordit.co/UNYQp7OWns
just added compute_sudo=True to related_bill_id

Comment (pedrobaeza) [reply account_invoice_inter_company/views/account_move_views.xml]: You will have user complaining about such error. And out of this, you don't know the company of the bill. I consider that info important.

Comment (ilyasProgrammer) [reply account_invoice_inter_company/views/account_move_views.xml]: Company of a bill you can see in the Customer field. Not company, but related partner.

Comment (pedrobaeza) [reply account_invoice_inter_company/views/account_move_views.xml]: Yes, true...

The access problem when you click is something to be fixed anyways.

Comment (ilyasProgrammer) [reply account_invoice_inter_company/views/account_move_views.xml]: Updated.

**Comment by pedrobaeza** [comment]
Seems OK. Can you squash?

**Comment by francesco-ooops** [comment]
@pedrobaeza go?

File: account_invoice_inter_company/views/account_move_views.xml [hunk 6]
```diff
@@ -13,4 +13,35 @@
...
         </field>
     </record>
+
+    <record id="view_move_form_inherit" model="ir.ui.view">
+        <field name="name">account_move_view_form_inherit</field>
+        <field name="model">account.move</field>
+        <field name="inherit_id" ref="account.view_move_form" />
+        <field name="arch" type="xml">
+            <notebook position="inside">
+                <page string="Intercompany Info" name="intercompany_info">
```
Comment (pedrobaeza) [inline account_invoice_inter_company/views/account_move_views.xml]: Sorry for not detecting this on time: please hide this tab if no related bill (which is the most common case).

Comment (ilyasProgrammer) [reply account_invoice_inter_company/views/account_move_views.xml]: ok

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**